src/ok.rs

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+use std::arch::x86_64;
+use std::mem::transmute;
+use std::str::from_utf8;
+
+#[derive(StructOpt)]
+pub enum SevGeneration {
+    #[structopt(about = "AMD SEV Extensions")]
+    Sev,
+
+    #[structopt(about = "AMD SEV-ES Extensions")]
+    Es,
+}
+
+struct CpuId {
+    pub name: &'static str,
+    pub leaf: u32,
+    pub func: fn(x86_64::CpuidResult) -> (bool, Option<String>),
```

Comment:
What's the difference between `(bool, Option<String>)` and `TestResult`? Could this just return a `TestResult`?

Reply:
The `Option<String>` is used to determine how to format the final string to be printed to the console in the `TestResult`

Reply:
Doesn't TestResult's message do that too?

Reply:
I came to the conclusion that `TestResult` really wasn't much different than `(bool, Option<String>)` yet added unnecessary complexity , so I removed it altogether.